### PR TITLE
authorization letter check in review step for lfng

### PR DESCRIPTION
**Pull request by urmikgov**

### CODE REVIEW

File: portal-frontend/src/app/features/applications/application-details/application-details.component.ts [hunk 1]
```diff
@@ -126,7 +126,10 @@ export class ApplicationDetailsComponent implements OnInit, OnDestroy {
 
       const isSelfApplicant = owners.length === 1 && this.primaryContact?.type.code === OWNER_TYPE.INDIVIDUAL;
       const isGovernmentContact = this.primaryContact?.type.code === OWNER_TYPE.GOVERNMENT;
-      this.needsAuthorizationLetter = !isGovernmentContact && !isSelfApplicant;
+      this.needsAuthorizationLetter = !isSelfApplicant;
```
Comment (trslater) [inline portal-frontend/src/app/features/applications/application-details/application-details.component.ts:129]: Consider replacing these 4 lines with the following:
```js
this.needsAuthorizationLetter = isGovernmentContact || !isSelfApplicant;
```
It is equivalent, but cleaner and easier to understand.

File: portal-frontend/src/app/features/notice-of-intents/notice-of-intent-details/notice-of-intent-details.component.ts [hunk 2]
```diff
@@ -124,7 +124,10 @@ export class NoticeOfIntentDetailsComponent implements OnInit, OnDestroy {
 
       const isSelfApplicant = owners.length === 1 && this.primaryContact?.type.code === OWNER_TYPE.INDIVIDUAL;
       const isGovernmentContact = this.primaryContact?.type.code === OWNER_TYPE.GOVERNMENT;
-      this.needsAuthorizationLetter = !isGovernmentContact && !isSelfApplicant;
+      this.needsAuthorizationLetter = !isSelfApplicant;
```
Comment (trslater) [inline portal-frontend/src/app/features/notice-of-intents/notice-of-intent-details/notice-of-intent-details.component.ts:127]: Same as above.

File: portal-frontend/src/app/features/applications/application-details/application-details.component.ts (hunk 1 above)
Comment (urmikgov) [reply portal-frontend/src/app/features/applications/application-details/application-details.component.ts]: gotcha, made changes

File: portal-frontend/src/app/features/notice-of-intents/notice-of-intent-details/notice-of-intent-details.component.ts (hunk 2 above)
Comment (urmikgov) [reply portal-frontend/src/app/features/notice-of-intents/notice-of-intent-details/notice-of-intent-details.component.ts]: done

**Review by trslater — APPROVED** [review]
LGTM 👍

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**